raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -48,16 +48,29 @@ public QuorumState(int localId,
         this.log = logContext.logger(QuorumState.class);
     }
 
-    public void initialize(OffsetAndEpoch offsetAndEpoch) throws IOException {
+    public void initialize(OffsetAndEpoch logEndOffsetAndEpoch) throws IOException {
         // We initialize in whatever state we were in on shutdown. If we were a leader
         // or candidate, probably an election was held, but we will find out about it
         // when we send Vote or BeginEpoch requests.
-        ElectionState election = store.readElectionState();
-        if (election.epoch < offsetAndEpoch.epoch) {
+
+        ElectionState election;
+        try {
+            election = store.readElectionState();
+        } catch (final IOException e) {
+            // For exceptions during state file loading (missing or not readable),
+            // we could assume the file is corrupted already and should be cleaned up.
+            log.info("Clear local quorum state store due to {}: {}", e, store.toString());
+            store.clear();
+
+            election = ElectionState.withUnknownLeader(logEndOffsetAndEpoch.epoch);
```

Comment:
Another way to handle this is to use
```java
election = ElectionState.withUnknownLeader(0);
```
Then we can skip the call to `writeElectionState` and rely on `becomeUnattachedFollower` after the `election.epoch < logEndOffsetAndEpoch.epoch` check below to update the state.

Reply:
Sounds good for simplification!